dvc/updater.py

```diff
@@ -40,36 +38,87 @@ def check(self):
+        self.latest = j['version']
+        open(self.updater_file, 'w+').close()
+
+    def _is_outdated(self):
+        l_major, l_minor, l_patch = [int(x) for x in self.latest.split('.')]
+        c_major, c_minor, c_patch = [int(x) for x in self.current.split('.')]
 
-        msg = (
+        return (l_major > c_major or
+                l_minor > c_minor or
+                l_patch > c_patch)
+
+    def _notify(self):
+        message = (
             'Update available {red}{current}{reset} -> {green}{latest}{reset}'
-            '\n'
-            'Run {yellow}pip{reset} install dvc {blue}--upgrade{reset}'
-            .format(
-                red=colorama.Fore.RED,
-                reset=colorama.Fore.RESET,
-                green=colorama.Fore.GREEN,
-                yellow=colorama.Fore.YELLOW,
-                blue=colorama.Fore.BLUE,
-                current=current,
-                latest=latest,
-            )
-        )
-
-        Logger.box(msg, border_color='yellow')
+            + '\n'
+            + self._get_update_instructions()
+        ).format(red=colorama.Fore.RED,
+                 reset=colorama.Fore.RESET,
+                 green=colorama.Fore.GREEN,
+                 yellow=colorama.Fore.YELLOW,
+                 blue=colorama.Fore.BLUE,
+                 current=self.current,
+                 latest=self.latest)
+
+        if sys.stdout.isatty():
+            Logger.box(message, border_color='yellow')
+
+    def _get_update_instructions(self):
+        instructions = {
+            'pip': 'Run {yellow}pip{reset} install dvc {blue}--upgrade{reset}',
+            'yum': 'Run {yellow}yum{reset} update {blue}dvc{reset}',
+            'yay': 'Run {yellow}yay{reset} {blue}-S{reset} dvc',
+            'brew': 'Run {yellow}brew{reset} upgrade {blue}dvc{reset}',
+            'apt': ('Run {yellow}apt-get{reset} install'
+                    ' {blue}--only-upgrade{reset} dvc'),
+            None: ('To upgrade follow this steps:\n'
+                   '1. Uninstall dvc binary\n'
+                   '2. Go to {blue}https://dvc.org{reset}\n'
+                   '3. Download and install new binary'),
+        }
+
+        package_manager = self._get_package_manager()
+
+        return instructions[package_manager]
+
+    def _get_package_manager(self):
+        package_managers = {
+            'rhel': 'yum',
+            'centos': 'yum',
+            'fedora': 'yum',
+            'amazon': 'yum',
+            'opensuse': 'yum',
+            'arch': 'yay',
+            'ubuntu': 'apt',
+            'debian': 'apt',
+            'darwin': 'brew',
+            'windows': None,
+        }
+
+        if self._is_installed_with_pip():
+            return 'pip'
+
+        return package_managers[distro.id()]
+
+    def _is_installed_with_pip(self):
+        command = ['pip', 'show', 'dvc']
+
+        try:
+            with open(os.devnull, 'w') as devnull:
+                return subprocess.check_call(command, stdout=devnull) == 0
```

Comment:
Just to make sure. If we install dvc with pip globally is it visible in virtualenv? If it's visible then this approach might fail.

Reply:
Good catch! I did a small test with Docker:

```bash
docker run -ti python bash

root@container/#: pip install dvc
root@container/#: mkdir /tmp/testing-pip && cd /tmp/testing-pip
root@container/#: pyhthon -m venv .venv && source .venv/bin/activate
root@container/#: pip list packages

# Just pip and setup tools :disappointed:
```

We need a smarter way to detect this.
Any suggestions, @efiop ?

Reply:
@mroutis my question about visibility had a little bit different meaning - can you actually access `dvc` installed globally at all while virt env is activated. If not, then there is nothing to worry about.

Reply:
@mroutis my point is that we don't need to run a special check for this if `dvc` is not visible inside virtualenv at all. Can you try to run it?

Reply:
@mroutis But your docker test demonstrates that, globally installed dvc is not visible in the venv, so we don't need to do any other checks there.

Reply:
@efiop, @shcheklein , it is not visible by `pip` but it still lives on the PATH.

The problem I'm seeing here is if you have `dvc` installed with `pip` and then you enter to your the virtual env (a pretty common approach), it will not detect a `pip` installation with the current `_is_installed_with_pip` method, as it uses the `pip` command directly and it will fallback to the system default, giving wrong instructions;

Reply:
@mroutis I agree. That is why we should use path heuristics instead, as discussed below and privately.